tools/codegen/api/translate.py

```diff
@@ -0,0 +1,143 @@
+from typing import Dict, Sequence, List, NoReturn
+from tools.codegen.api.types import *
+
+# This file implements a small program synthesis engine that implements
+# conversions between one API to another.
+#
+# The key data type in this file in CType, short for C++ semantic type.  A CType
+# represents a C++ type, plus semantic information about what it represents.
+# For example, consider the argument "bool pin_memory"; its normal C++ type is
+# "bool", but its C++ semantic type also keeps track that this represents a
+# "pin_memory"; you can't just use a random other boolean in a context where you
+# need a "pin_memory"!
+#
+# The translator takes a list of needed CTypes, and then figures out how
+# to construct expressions with these CTypes from the given bindings.  Many
+# of these expressions are trivial (I need a Tensor other; there's a Tensor
+# other scope); others are more nontrivial and may require packing/unpacking.
+# Some examples of non-trivial action:
+#
+#   - Need the "dtype" binding?  Well, maybe "dtype" isn't available
```

Comment:
Gathering isn't be necessary anymore since all ops with TensorOptions are already c10-full. You should only need scattering in some situations

Reply:
I'll try deleting the gather rule and see if it works.